store/CSDK/test/main.cpp

```diff
@@ -645,6 +653,278 @@ bool testWriteData(JNIEnv *env, char *path, int argc, char *argv[]) {
+
+            char ctrFloat[10];
+            gcvt(floatValue, 10, ctrFloat);
+            floatValue = floatValue + 2;
+            jobject floatField = env->NewStringUTF(ctrFloat);
+            env->SetObjectArrayElement(arr, 8, floatField);
+
+            jobject arrayField = env->NewStringUTF("Hello#World#From#Carbon");
+            env->SetObjectArrayElement(arr, 9, arrayField);
+
+
+            writer.write(arr);
+
+            env->DeleteLocalRef(stringField);
+            env->DeleteLocalRef(shortField);
+            env->DeleteLocalRef(intField);
+            env->DeleteLocalRef(longField);
+            env->DeleteLocalRef(doubleField);
+            env->DeleteLocalRef(floatField);
+            env->DeleteLocalRef(dateField);
+            env->DeleteLocalRef(timeField);
+            env->DeleteLocalRef(boolField);
+            env->DeleteLocalRef(arrayField);
+            env->DeleteLocalRef(arr);
+        }
+        writer.close();
+
+        CarbonReader carbonReader;
+        carbonReader.builder(env, path);
+        carbonReader.build();
+        int i = 0;
+        int printNum = 10;
+        CarbonRow carbonRow(env);
+        while (carbonReader.hasNext()) {
+            jobject row = carbonReader.readNextRow();
+            i++;
+            carbonRow.setCarbonRow(row);
+            if (i < printNum) {
+                printf("%s\t%d\t%ld\t", carbonRow.getString(0), carbonRow.getInt(1), carbonRow.getLong(2));
+                jobjectArray array1 = carbonRow.getArray(3);
+                jsize length = env->GetArrayLength(array1);
+                int j = 0;
+                for (j = 0; j < length; j++) {
+                    jobject element = env->GetObjectArrayElement(array1, j);
+                    char *str = (char *) env->GetStringUTFChars((jstring) element, JNI_FALSE);
+                    printf("%s\t", str);
+                }
+                printf("%d\t", carbonRow.getShort(4));
+                printf("%d\t", carbonRow.getInt(5));
+                printf("%ld\t", carbonRow.getLong(6));
+                printf("%lf\t", carbonRow.getDouble(7));
+                bool bool1 = carbonRow.getBoolean(8);
+                if (bool1) {
+                    printf("true\t");
+                } else {
+                    printf("false\t");
+                }
+                printf("%f\t\n", carbonRow.getFloat(9));
+            }
+            env->DeleteLocalRef(row);
+        }
+        carbonReader.close();
+    } catch (jthrowable ex) {
+        env->ExceptionDescribe();
+        env->ExceptionClear();
+    }
+}
+
+/**
+  * test WithTableProperties interface
+  *
+  * @param env  jni env
+  * @param path file path
+  * @param argc argument counter
+  * @param argv argument vector
+  * @return true or throw exception
+  */
+bool testWithTableProperty(JNIEnv *env, char *path, int argc, char **argv) {
+
+    char *jsonSchema = "[{stringField:string},{shortField:short}]";
+    try {
```

Comment:
same as above

Reply:
this one is better to keep. It need add random if add this to testWriteData, keep this will more simply to understand for user.